src/hotspot/share/gc/g1/g1_globals.hpp

```diff
@@ -302,7 +302,12 @@
           "of getloadavg() at which G1 triggers a periodic GC. A load "     \
           "above this value cancels a given periodic GC. A value of zero "  \
           "disables this check.")                                           \
-          range(0.0, (double)max_uintx)
+          range(0.0, (double)max_uintx)                                     \
+                                                                            \
+  product(bool, G1AllowProactiveGC, true,                                   \
```

Comment:
Imho this flag should be either an experimental or diagnostic flag. The reason is that it is only supposed to be used exclusively for troubleshooting, not giving generic guidelines for the collector (like e.g. MaxHeapSize).

This also avoids a CSR for this change.

Reply:
Thanks @tschatzl, that makes sense to me. I'll push an update shortly that makes it a "diagnostic" flag.